plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetastoreModule.java

```diff
@@ -15,17 +15,45 @@
 
 import com.google.inject.Binder;
 import com.google.inject.Module;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
 import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.MetastoreTypeConfig;
 import io.trino.plugin.hive.metastore.cache.CachingHiveMetastore;
+import io.trino.plugin.hive.metastore.cache.CachingHiveMetastoreModule;
+import io.trino.plugin.hive.metastore.cache.ForCachingHiveMetastore;
+import io.trino.plugin.hive.metastore.file.FileMetastoreModule;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastoreModule;
 
 import javax.inject.Inject;
 
+import java.util.Optional;
+
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+import static java.util.Objects.requireNonNull;
+
 public class IcebergMetastoreModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
+    private final Optional<HiveMetastore> metastore;
+
+    public IcebergMetastoreModule(Optional<HiveMetastore> metastore)
+    {
+        this.metastore = requireNonNull(metastore, "metastore is null");
+    }
+
     @Override
-    public void configure(Binder binder)
+    protected void setup(Binder binder)
     {
+        if (metastore.isPresent()) {
+            binder.bind(HiveMetastore.class).annotatedWith(ForCachingHiveMetastore.class).toInstance(metastore.get());
+            install(new CachingHiveMetastoreModule());
+        }
+        else {
+            bindMetastoreModule("thrift", new ThriftMetastoreModule());
+            bindMetastoreModule("file", new FileMetastoreModule());
+            // TODO bindMetastoreModule("glue", new GlueMetastoreModule());
```

Comment:
Remove this. I expect we will implement Glue using a catalog rather than through the Hive metastore API.